modules/codegen/src/smithy4s/codegen/internals/ModelLoader.scala

```diff
@@ -74,9 +74,14 @@ private[codegen] object ModelLoader {
       }
     }
 
+    val validatorClassLoader = locally {
+      val jarUrls = deps.map(_.toURI().toURL()).toArray
+      new URLClassLoader(jarUrls, currentClassLoader)
+    }
+
     // Loading the upstream model
     val upstreamModel = Model
-      .assembler()
+      .assembler(validatorClassLoader)
```

Comment:
This should make the classes (validators, trait services) visible to the assembler - but notably not making the models visible.

Reply:
notably this doesn't solve the problems that #1705 and #1707 are attempting to solve, but it's a step in the right direction and doesn't clash with them